File: problems/symmetric_tree/symmetric_tree.py
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        return self.isSymmetricHelper(root.left, root.right)

    def isSymmetricHelper(
        self, one: Optional[TreeNode], two: Optional[TreeNode]
    ) -> bool:
        if not one or not two:
            return one == two

        return (
            one.val == two.val
            and self.isSymmetricHelper(one.left, two.right)
            and self.isSymmetricHelper(one.right, two.left)
        )

File: problems/symmetric_tree/SymmetricTree.java
package problems.symmetric_tree;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }
}

public class SymmetricTree {
    public boolean isSymmetric(TreeNode root) {

        if (root == null)
            return true;

        return helper(root.left, root.right);
    }

    private boolean helper(TreeNode one, TreeNode two) {
        if (one == null || two == null)
            return one == two;

        return one.val == two.val && helper(one.left, two.right) && helper(one.right, two.left);
    }
}
